src/Symfony/Component/Serializer/Encoder/CsvEncoder.php

```diff
@@ -154,7 +154,7 @@ public function decode(string $data, string $format, array $context = []): mixed
                     $headerCount = array_fill(0, $nbCols, 1);
                 } else {
                     foreach ($cols as $col) {
-                        $header = explode($keySeparator, $col);
+                        $header = explode($keySeparator, $col ?? '');
```

Comment:
Can you add a test to cover this behavior?

Reply:
I added the test case while merging